employees/static/employees/scripts/modalsFunctionality.js

```diff
@@ -26,8 +33,44 @@ $(function() {
         document.activeElement.blur();
     });
 
-    $("#modal").on('shown.bs.modal', function() {
+    $("#modal").on('show.bs.modal', function (){
+        let reportForm = $(this).find("form")
+        if (reportForm.attr("name") === "create-report-form") {
+            getFieldsValuesFromLocalStorage(reportForm);
+        }
+
+    }).on('shown.bs.modal', function() {
         $("body").removeClass("modal-open");
         $(".custom-modal").removeClass("modal");
+
+    }).on('hide.bs.modal', function() {
+        let reportForm = $(this).find("form")
+        if (reportForm.attr("name") === "create-report-form"){
+            passFormValuesToLocalStorage(reportForm);
+        }
     });
+
+    resetLocalStorage();
 });
+
+function passFormValuesToLocalStorage(form) {
+    var createReportFields = {}
+    for (let[formField, formFieldSelector] of Object.entries(formFieldsSelectors)){
+        createReportFields[formField] = form.find(formFieldSelector).val()
+    }
+    localStorage.setItem("createReportFields", JSON.stringify(createReportFields));
+}
+
+function getFieldsValuesFromLocalStorage(form) {
+    if (localStorage.getItem("createReportFields") !== "") {
+        var storedCreateReportFields = JSON.parse(localStorage.getItem("createReportFields"));
+
+        for (let[formField, formFieldSelector] of Object.entries(formFieldsSelectors)){
+            form.find(formFieldSelector).val(storedCreateReportFields[formField]);
+        }
+    }
+}
+
+function resetLocalStorage() {
+    localStorage.setItem("createReportFields", "");
```

Comment:
Instead of setting again some item better to use `removeItem`?

Reply:
Done